### PR TITLE
Added LinkedInAuthenticationOptions.Proxy host for web applications r…

### PR DESCRIPTION
**Pull request by hokukim**
…unning behind a proxy. This allows the middleware to authenticate for the proxy request host instead of the application request host.

### CODE REVIEW

File: src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs [hunk 1]
```diff
@@ -271,5 +271,15 @@ protected override Task ApplyResponseChallengeAsync()
...
         }
+
+        /// <summary>
+        ///     Gets proxy host name from <see cref="LinkedInAuthenticationOptions"/> if it is set.
+        ///     If proxy host name is not set, gets application request host name.
+        /// </summary>
+        /// <returns>Host name.</returns>
+        private string GetHostName()
+        {
+            return string.IsNullOrWhiteSpace(Options.ProxyHost) ? Request.Host.ToString() : Options.ProxyHost;
```
Comment (TerribleDev) [inline src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs:282]: don't proxies add the origin host as an `X-Forwarded-Host` header?

Comment (hokukim) [reply src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs:282]: Thanks for the suggestion :) I'll try that option with our proxy host.

Comment (hokukim) [reply src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs:282]: Unfortunately the X-Forwarded-Host header suggestion did not work for us. So we are using this solution in the meantime.

Comment (TerribleDev) [reply src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs:282]: Alrighty thanks @hokukim I'll merge this in tomorrow-ish

Comment (hokukim) [reply src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs:282]: Great, thank you!

**Pull request merged**